### PR TITLE
Middlewares refactor

### PR DESCRIPTION
**Pull request by ianaya89**
This PR improves the middlewares API, it changes the methods names to `before`, `success`,  `error` and adds the `after` method. The error handler has been redesign to execute an array of functions, not returning any promise rejection.

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -419,4 +411,53 @@ describe('HTTP -> http', () => {
...
+    });
+
+    it('makes a GET request to baseURL + path using error and after middlewares', () => {
+      function after(res) {
+        res.after = true;
+        return Promise.resolve(res);
+      }
+
+      function error(err) {
+        err.success = true;
```
Comment (ndelvalle) [inline test/index.spec.js]: `err.error = true` I think?

**Comment by ndelvalle** [comment]
Im not sure about not returning promises in the `error` middlewares but I don't have a strong position at the moment, so LGTM 👍  Thanks for the PR!

**Comment by gillchristian** [comment]
LGTM :+1: 

I like the new names, make much more sense :sunglasses: 

As @ndelvalle says, we should find a better way to handle `error`s, but this surely goes in the right direction.

Since you added `node-inspector`, we could add an `npm script` to run it.

**Comment by ianaya89** [comment]
I have same feelings about `error` handling but we can figure out a better solution while we test it in real applications. I will be using this in some prod apps, so would be good test case to look of better solutions.

About `node-inspector`, I added to make it work with jest but I couldn't yet. I will try it later, in the meantime, you can remove the dep if you want.

**Comment by gillchristian** [comment]
Seems like a good path to take to me.

Already published a [new version](https://github.com/Huemul/trae/releases/tag/v0.0.10) so you can use it right away.

No worries about `node-inspector`.

**Pull request merged**